<!DOCTYPE html>
<html lang="en">
  <head>
    <title>WEBPAGE DESIGN</title>
    <link rel="stylesheet" href="style.css" />
    <script src="myScript1.js" defer></script>
    <link
      href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css"
      rel="stylesheet"
      integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3"
      crossorigin="anonymous"
    />
    <script
      src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js"
      integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p"
      crossorigin="anonymous"
    ></script>
  </head>
  <section class="firstsection">
    <div class="box-main">
      <div class="firstHalf">
        <h1 class="text-biggg" id="web"><u>Exceptional Handling</u></h1>
        <h3 class="text-big" id="web"><u>User Defined Exception</u></h3>

        <p class="text-small">
          An exception is an issue (run time error)
          that occurred during the execution of a
          program. When an exception occurred the 
          program gets terminated abruptly and,
          the code past the line that generated 
          the exception never gets executed.
          Java provides us the facility to create
          our own exceptions which are basically 
          derived classes of Exception. 
          Creating our own Exception is known as a
          custom exception or user-defined exception. 
          Basically, Java custom exceptions are used to
          customize the exception according to user needs.
          In simple words, we can say that a User-Defined 
          Exception or custom exception is creating your 
          own exception class and throwing that exception
           using the ‘throw’ keyword.
        </p>

        <div style="margin: left 30px" class="card-body">
            <h5 class="card-title">Example-1</h5>
            <h6 class="card-subtitle mb-2 text-muted">
              //Java Program to illustrate the use of finally
            </h6>
          </div>
          <div class="card" style="width: 60rem">
            <div style="margin-left: -40px" class="card-body">
              <pre class="lead"><br>
                // A Class that represents use-defined exception
 
class MyException extends Exception {
    public MyException(String s)
    {
        // Call constructor of parent Exception
        super(s);
    }
}
 
// A Class that uses above MyException
public class Main {
    // Driver Program
    public static void main(String args[])
    {
        try {
            // Throw an object of user defined exception
            throw new MyException("GeeksGeeks");
        }
        catch (MyException ex) {
            System.out.println("Caught");
 
            // Print the message from MyException object
            System.out.println(ex.getMessage());
        }
    }
}


                    <br>
                    <b><u>OUTPUT</u></b>
                    Something went wrong.
                    The 'try catch' is finished
              <br>
              </pre>
              <div class="ex">
                <img class="output" src="images/output.png" alt="" />
              </div>
            </div>
          </div>
        </div>
      </div>
      <br><br>
        <div class="card-footer text-muted">
          <b>Ayush Raj Sharma, 2029135</b> <br>
           GROUP_11
       </div>
    </div>
  </section>
</html>
